Source/Cake.Email/EmailProvider.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Net;
+using System.Net.Mail;
+using System.Text;
+using Cake.Core;
+using Cake.Core.Annotations;
+
+namespace Cake.Email
+{
+    /// <summary>
+    /// Contains functionality related to emails
+    /// </summary>
+    [CakeAliasCategory("Email")]
+    public sealed class EmailProvider
+    {
+        private readonly ICakeContext _context;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EmailProvider"/> class.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        public EmailProvider(ICakeContext context)
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        /// Sends an email via the SendGrid API, based on the provided settings
+        /// </summary>
+        /// <param name="senderName">The name of the person sending the email</param>
+        /// <param name="senderAddress">The email address of the person sending the email</param>
+        /// <param name="recipientName">The name of the person who will receive the email</param>
+        /// <param name="recipientAddress">The email address of the person who will recieve the email</param>
+        /// <param name="subject">The subject line of the email</param>
+        /// <param name="content">The body of the email</param>
+        /// <param name="settings">The settings to be used when sending the email</param>
+        /// <returns>An instance of <see cref="EmailResult"/> indicating success/failure</returns>
+        public EmailResult Send(string senderName, string senderAddress, string recipientName, string recipientAddress, string subject, string content, EmailSettings settings)
```

Comment:
From a consistency point of view, in the other two providers, you have called this method `SendEmail`.  Does it make sense to keep it the same here?